lib/url.js

```diff
@@ -1020,10 +1020,10 @@ Url.prototype.parseHost = function parseHost() {
 // When used internally, we are not obligated to associate TypeError with
 // this function, so non-strings can be rejected by underlying implementation.
 // Public API has to validate input and throw appropriate error.
-function pathToFileURL(path) {
+function pathToFileURL(path, resolveBase) {
   validateString(path, 'path');
 
-  return _pathToFileURL(path);
+  return _pathToFileURL(path, resolveBase);
```

Comment:
Let's not do that in this PR, it would need to be its own discussion/PR

Reply:
This is required or the whole things doesn't work. I agree we need to do is separatly, but there is no way other than duplicating all that logic in `pathFoFileURL` in here.

Reply:
Unless you know of a better way?

Reply:
Hum maybe I'm missing something, but I don't see why it's needed, AFAIK if you put

Reply:
I made this change, and it passes my tests so 👍. I read the code for `pathToFileURL` and was pretty sure this was not going to work with all cases but maybe I am wrong. We have nowhere near enough test coverage yet, so I still don't feel like I have a clear grasp of file path to URL stuff in a cross platform way but 🤷 I will open the PR without this change.